templates/pre-commit.protect-master

```diff
@@ -19,12 +20,25 @@ fi
 # Get the current branch name
 CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
 
+# Check if the current branch is `master` or `main`
+case "$CURRENT_BRANCH" in
+    master|main)
+        echo "FATAL ERROR: You cannot commit directly to the $CURRENT_BRANCH branch."
+        exit 1
+        ;;
+esac
+
+# Exit if it's a merge commit
+if [ "$(git rev-list --parents -n 1 HEAD | wc -w)" -gt 2 ]; then
+    exit 0
+fi
+
 # Get the known deploy branches from the git config (comma-separated list)
 KNOWN_DEPLOY_BRANCHES=$(git config --get ka.olc.targetBranches)
```

Comment:
Just a style thing, but I'd move this down after line 46, so that the two cannot-commit checks are together.

Reply:
I cleaned it up 😄